src/components/Pages/PlaylistCollection/Container.js

```diff
@@ -45,14 +45,16 @@ const PlaylistCollectionContainer = ({ cId, t, cuId }) => {
 
       // Fetch full units data if needed.
       if (Array.isArray(cuIDs) && cuIDs.length > 0) {
-        cuIDs.forEach(cuID => {
-          if (!fullUnitFetchedMap[cuID] && !wipMap.units[cuID] && !errorMap.units[cuID]) {
-            const cu = content_units.find(x => x.id === cuID);
-            if (!cu || !cu.files || !cu.tags || !cu.sources) {
-              dispatch(actions.fetchUnit(cuID));
-            }
-          }
+        const cusForFetch = cuIDs.filter(cuID => {
+          if (fullUnitFetchedMap[cuID] || wipMap.units[cuID] || errorMap.units[cuID])
+            return false;
+          const cu = content_units.find(x => x.id === cuID);
+          return !cu?.files;
```

Comment:
@bbfsdev I remove check tags and sources from here, look like its not need

Reply:
That is not true it solves this issue:
https://issues.kbb1.com/issue/AR-874

Does this causes the problem?

Reply:
is issue is about played now CU right? So why you need fetch data for all units of collection?